lib/swim.ml

```diff
@@ -0,0 +1,127 @@
+open Base.Hashtbl
+
+type message =
```

Comment:
Given how Bin_prot encodes sum types, we should probably find a way to differentiate SWIM messages from the consumer's custom defined messages. One way to do this is to prepend some header bytes to the payload when packing SWIM messages and warn the consumer about this. What do you think?